icu4c/source/i18n/units_converter.cpp

```diff
@@ -212,8 +213,8 @@ Factor loadSingleFactor(StringPiece source, const ConversionRates &ratesInfo, UE
         return {};
     }
 
-    Factor result = extractFactorConversions(conversionUnit->factor.toStringPiece(), status);
-    result.offset = strHasDivideSignToDouble(conversionUnit->offset.toStringPiece(), status);
+    Factor result = extractFactorConversions(conversionUnit->factor.data(), status);
+    result.offset = strHasDivideSignToDouble(conversionUnit->offset.data(), status);
```

Comment:
it looks like we are going back to passing strings with known lengths as NUL-terminated pointers and calling strlen() again to make the implicit string_view...